File: 02_ProgrammingFundamentals/_06_ObjectsAndClasses/_01_Lecture/_05_Students.java
package _02_ProgrammingFundamentals._06_ObjectsAndClasses._01_Lecture;

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class _05_Students {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input value
        String console = scan.nextLine();

        // Add variable
        List<Student> studentList = new ArrayList<>();

        // Save a list of students and their information
        while (!console.equals("end")) {
            String[] lineFromConsole = console.split("\\s+");

            String firstName = lineFromConsole[0];
            String secondName = lineFromConsole[1];
            int age = Integer.parseInt(lineFromConsole[2]);
            String town = lineFromConsole[3];

            Student student = new Student(firstName, secondName, age, town);
            studentList.add(student);

            console = scan.nextLine();
        }

        String consoleTown = scan.nextLine();

        // Print result
        for (Student element : studentList) {
            if (element.getTown().equals(consoleTown)) {
                System.out.println(element);
            }
        }

        scan.close();
    }
}

// Create practice Student class
class Student {
    private String firstName;
    private String secondName;
    private int age;
    private String town;

    public Student(String firstName, String secondName, int age, String town) {
        this.firstName = firstName;
        this.secondName = secondName;
        this.age = age;
        this.town = town;
    }

    public String getTown() {
        return this.town;
    }

    @Override
    public String toString() {
        return this.firstName + " " + this.secondName + " is " + this.age + " years old";
    }
}

File: 24_C#Fundamentals/_06_ObjectsAndClasses/_01_Lecture/_05_Students/_05_Students.cs
using System;

public class _05_Students
{
    public static void Main(string[] args)
    {
        // Read input value
        string console = Console.ReadLine();

        // Add variable
        List<Student> studentList = new List<Student>();

        // Save a list of students and their information
        while (!console.Equals("end"))
        {
            string[] lineFromConsole = console.Split(" ");

            string firstName = lineFromConsole[0];
            string secondName = lineFromConsole[1];
            int age = int.Parse(lineFromConsole[2]);
            string town = lineFromConsole[3];

            Student student = new Student(firstName, secondName, age, town);
            studentList.Add(student);

            console = Console.ReadLine();
        }

        string consoleTown = Console.ReadLine();

        // Print result
        foreach (Student element in studentList)
        {
            if (element.Town.Equals(consoleTown))
            {
                Console.WriteLine(element);
            }
        }
    }
}

// Create practice Student class
class Student
{
    public string FirstName { get; }
    public string SecondName { get; }
    public int Age { get; }
    public string Town { get; }

    public Student(string firstName, string secondName, int age, string town)
    {
        this.FirstName = firstName;
        this.SecondName = secondName;
        this.Age = age;
        this.Town = town;
    }

    public override string ToString()
    {
        return $"{this.FirstName} {this.SecondName} is {this.Age} years old.";
    }
}
